chemprop/models/model.py

```diff
@@ -255,3 +248,22 @@ def load_from_file(cls, model_path, map_location=None, strict=True) -> MPNN:
         model.load_state_dict(state_dict, strict=strict)
 
         return model
+
+
+def load_submodules(checkpoint_path, **kwargs):
+    hparams = torch.load(checkpoint_path)["hyper_parameters"]
+
+    if "blocks" in hparams["message_passing"]:
+        mp_hparams = hparams["message_passing"]
+        mp_hparams["blocks"] = [
+            block_hparams.pop("cls")(**block_hparams) for block_hparams in mp_hparams["blocks"]
+        ]
+        message_passing = mp_hparams.pop("cls")(**mp_hparams)
+        kwargs["message_passing"] = message_passing
```

Comment:
this seems somewhat error prone. What if the `hyper_parameters` key for an `MPNN` contains `"blocks"`? Reasonably, we should ignore this, but sharing this code between the two classes necessitates the _assumption_ of its structure.

Reply:
I agree this isn't ideal. If we left `load_submodules()` outside of `MPNN` we could bring back `load_from_checkpoint()` in `MulticomponentMPNN` and put a similar function in `multi.py`. Or if we make `load_submodules()` a classmethod of `MPNN`, then `MulticomponentMPNN` could overload it. The only catch is each needs to be self sufficient because we can't call `super()`. 
In MPNN
```
    @classmethod
    def load_submodules(cls, checkpoint_path, **kwargs):
        hparams = torch.load(checkpoint_path)["hyper_parameters"]

        kwargs |= {
            key: hparams[key].pop("cls")(**hparams[key])
            for key in ("message_passing", "agg", "predictor")
            if key not in kwargs
        }
        return kwargs
```
In MulticomponentMPNN
```
    @classmethod
    def load_submodules(cls, checkpoint_path, **kwargs):
        hparams = torch.load(checkpoint_path)["hyper_parameters"]

        hparams["message_passing"]["blocks"] = [
            block_hparams.pop("cls")(**block_hparams)
            for block_hparams in hparams["message_passing"]["blocks"]
        ]
        kwargs |= {
            key: hparams[key].pop("cls")(**hparams[key])
            for key in ("message_passing", "agg", "predictor")
            if key not in kwargs
        }
        return kwargs
```
This works for loading both `MPNN` and `MulticomponentMPNN` models when `load_from_checkpoint()` is class method of `MPNN` that calls `cls.load_submodules()`. What do you think?

Reply:
I like the second option better. Obviously there's some duplicate code, but it's only across one class and it's not strictly worse style-wise either. In this case, a user doesn't have to bark up the class hierarchy to see what the function does, as it's all contained in one place